agenta-backend/agenta_backend/main.py

```diff
@@ -12,12 +23,34 @@
     "http://0.0.0.0:3001",
 ]
 
+
+@asynccontextmanager
+async def lifespan(application: FastAPI):
+    tags_data = await retrieve_templates_from_dockerhub_cached()
```

Comment:
We are. When the application initiates, we make an API call to Docker Hub to retrieve the templates. What the `retrieve_templates_from_dockerhub_cached` function does is:

- checks for a cache and return it if we do
- otherwise, make a call to the docker hub, cache the response and return the response

Whenever the application restarts, instead of making another call to Docker Hub, we get the cache response.

cc: @mmabrouk